app/views/nsm/work_items/index.html.erb

```diff
@@ -19,27 +19,54 @@
-  <% head = Nsm::V1::WorkItem.headers %>
-  <% work_items.group_by(&:completed_on).each do |completed_on, work_items_for_date| %>
-    <%=
-      rows = work_items_for_date.map do |work_item|
-        link = if claim.editable_by?(current_user)
-                  link_to(t('.change'), edit_nsm_claim_work_item_path(claim_id: claim.id, id: work_item.id), data: { turbo: 'false' })
-                else
-                  link_to(t('.view'), nsm_claim_work_item_path(claim_id: claim.id, id: work_item.id), data: { turbo: 'false' })
-                end
-        [
-          *work_item.table_fields,
-          link
-        ]
-      end
-
-      govuk_table do |table|
-        table.with_caption(text: completed_on.strftime('%-d %B %Y'), size: 's')
-        table.with_head(rows: [head])
-        table.with_body(rows: rows, first_cell_is_header: true)
-      end
-    %>
-  <% end %>
+  <table class="govuk-table" data-module="moj-sortable-table" aria-describedby="page-title">
+    <thead class="govuk-table__head">
+      <tr class="govuk-table__row">
+        <% Nsm::V1::WorkItem.headers.each_with_index do |(column_key, additional_classes), index| %>
+          <%= table_header(column_key,
+                          "nsm.work_items.index",
+                          index,
+                          nsm_claim_work_items_path(claim),
+                          @sort_by,
+                          @sort_direction,
+                          additional_classes: ) %>
+        <% end %>
+      </tr>
+    </thead>
+
+    <tbody class="govuk-table__body app-task-list__items">
+      <% work_items.each do |work_item| %>
+        <tr class="govuk-table__row app-task-list__item">
+          <th class="govuk-table__header"><%= work_item.position %></th>
+          <td class="govuk-table__cell">
+            <%=
+              if claim.editable_by?(current_user)
+                link_to(work_item.work_type, edit_nsm_claim_work_item_path(claim_id: claim.id, id: work_item.id), data: { turbo: 'false' })
+              else
+                link_to(work_item.work_type, nsm_claim_work_item_path(claim_id: claim.id, id: work_item.id), data: { turbo: 'false' })
+              end
+            %>
+          </td>
+
+          <td class="govuk-table__cell"><%= work_item.formatted_completed_on %></td>
+          <td class="govuk-table__cell"><%= work_item.fee_earner %></td>
+
+          <td class="govuk-table__cell govuk-table__cell--numeric">
+            <%= work_item.formatted_time_spent %>
+          </td>
+          <td class="govuk-table__cell govuk-table__cell--numeric">
+            <%= work_item.formatted_uplift %>
+          </td>
+          <td class="govuk-table__cell govuk-table__cell--numeric">
+            <%= work_item.formatted_requested_amount %>
+          </td>
+          <td class="govuk-table__cell govuk-table__cell--numeric">
+            <%= work_item.formatted_allowed_amount %>
+          </td>
+        </tr>
+      <% end %>
+    </tbody>
+  </table>
+
```

Comment:
You can try using govuk_table component here. I haven't tried it yet, just a suggestion.
```ruby
<%=
  govuk_table(html_attributes: { class: "govuk-table", 'data-module': "moj-sortable-table", 'aria-describedby': "page-title" }) do |table|
    # table head
    table.with_head(rows: [
      Nsm::V1::WorkItem.headers.map do |(column_key, additional_classes), index|
        {
          text: table_header(column_key, "nsm.work_items.index", index, nsm_claim_work_items_path(claim), @sort_by, @sort_direction, additional_classes: additional_classes),
          classes: additional_classes
        }
      end
    ])

    # table body
    table.with_body(rows: work_items.map do |work_item|
      [
        { text: work_item.position, header: true },
        {
          text: if claim.editable_by?(current_user)
                  link_to(work_item.work_type, edit_nsm_claim_work_item_path(claim_id: claim.id, id: work_item.id), data: { turbo: 'false' })
                else
                  link_to(work_item.work_type, nsm_claim_work_item_path(claim_id: claim.id, id: work_item.id), data: { turbo: 'false' })
                end
        },
        { text: work_item.formatted_completed_on },
        { text: work_item.fee_earner },
        { text: work_item.formatted_time_spent, numeric: true },
        { text: work_item.formatted_uplift, numeric: true },
        { text: work_item.formatted_requested_amount, numeric: true },
        { text: work_item.formatted_allowed_amount, numeric: true }
      ]
    end)
  end
%>
```

Reply:
I guess at this point I don;t feel like the helper is helping... that is why I switched back to using html. if it is already needing tio be this complex it most likely a smell that we are doing it wrong. :)